lib/email-validation/blacklisted_email.rb

```diff
@@ -16,19 +12,13 @@ class BlacklistedEmail < ::ActiveRecord::Base
 
     before_save :parse_email
 
-    workflow do
-      state :bounce
-      state :unsubscribe
-      state :admin
-    end
-
     ORIGIN_ADMIN = 'admin'
     ORIGIN_BOUNCE = 'bounce'
     ORIGIN_UNSUBSCRIBE = 'unsubscribe'
 
-    def self.exists? email
+    def self.bounce_or_admin? email
       email = BlacklistedEmail.parse_email_address(email.to_s)
-      BlacklistedEmail.where(email: email).exists?
+      BlacklistedEmail.where(email: email, origin: [ORIGIN_BOUNCE, ORIGIN_ADMIN]).exists?
```

Comment:
as we discussed, let's wait with such changes once we re-think the whole functionality - it turned out we have to do it anyway.